examples/flax/language-modeling/run_clm_flax.py

```diff
@@ -0,0 +1,632 @@
+            labels = batch.pop("labels")
+            logits = state.apply_fn(**batch, params=params, dropout_rng=dropout_rng, train=True)[0]
+            loss = loss_fn(logits, labels)
+            return loss
+
+        grad_fn = jax.value_and_grad(compute_loss)
+        loss, grad = grad_fn(state.params)
+        if distributed:
+            grad = jax.lax.pmean(grad, "batch")
+
+        new_state = state.apply_gradients(grads=grad)
+
+        metrics = {"loss": loss, "learning_rate": linear_decay_lr_schedule_fn(state.step)}
+        if distributed:
+            metrics = jax.lax.pmean(metrics, axis_name="batch")
+
+        return new_state, metrics, new_dropout_rng
+
+    # Define eval fn
+    def _eval_step(params, batch):
+        labels = batch.pop("labels")
+        logits = model(**batch, params=params, train=False)[0]
+        loss = loss_fn(logits, labels)
+
+        # summarize metrics
+        metrics = {"loss": loss}
+        if distributed:
+            metrics = jax.lax.pmean(metrics, axis_name="batch")
+        return metrics
+
+    # Create parallel version of the train and eval step
+    if distributed:
+        train_step = jax.pmap(_train_step, "batch", donate_argnums=(0,))
+        eval_step = jax.pmap(_eval_step, "batch", donate_argnums=(0,))
+
+        # Replicate the train state on each device
+        state = jax_utils.replicate(state)
+    else:
+        train_step = jax.jit(_train_step, donate_argnums=(0,))
+        eval_step = jax.jit(_eval_step)
+
+    logger.info("***** Running training *****")
+    logger.info(f"  Num examples = {len(train_dataset)}")
+    logger.info(f"  Num Epochs = {num_epochs}")
+    logger.info(f"  Instantaneous batch size per device = {training_args.per_device_train_batch_size}")
+    logger.info(f"  Total train batch size (w. parallel, distributed & accumulation) = {train_batch_size}")
+    logger.info(f"  Total optimization steps = {total_train_steps}")
+
+    train_metrics = []
+    train_time = 0
+    epochs = tqdm(range(num_epochs), desc=f"Epoch ... (1/{num_epochs})", position=0)
+    for epoch in epochs:
+        # ======================== Training ================================
+        train_start = time.time()
+
+        # Create sampling rng
+        rng, input_rng = jax.random.split(rng)
+
+        # Generate an epoch by shuffling sampling indices from the train dataset
+        train_loader = data_loader(input_rng, train_dataset, train_batch_size, shuffle=True, distributed=distributed)
+        steps_per_epoch = len(train_dataset) // train_batch_size
+        # train
+        for _ in tqdm(range(steps_per_epoch), desc="Training...", position=1, leave=False):
+            batch = next(train_loader)
+            state, train_metric, dropout_rngs = train_step(state, batch, dropout_rngs)
+            train_metrics.append(train_metric)
+
+        train_time += time.time() - train_start
+
+        if distributed:
+            train_metric = unreplicate(train_metric)
```

Comment:
I believe you can use `get_metrics` here like you do for `eval`.

Reply:
this is a single `dict` for last train step in the epoch, so we can't use `get_metrics` here.